frontend/src/features/comparisonSeries/ComparisonSeries.tsx

```diff
@@ -50,14 +52,20 @@ const ComparisonSeries = ({
   const [refreshLeft, setRefreshLeft] = useState(false);
   // a limited list of entities that can be used to suggest new comparisons
   const [alternatives, setAlternatives] = useState<Array<Entity>>([]);
-  // a list of already made comparisons, allowing to not suggest two times the
-  // same comparison to a user
+  // an array of already made comparisons, allowing to not suggest two times the
+  // same comparison to a user, formatted like this ['uidA/uidB', 'uidA/uidC']
   const [comparisonsMade, setComparisonsMade] = useState<Array<string>>([]);
 
   const searchParams = new URLSearchParams(location.search);
   const uidA: string = searchParams.get(UID_PARAMS.vidA) || '';
   const uidB: string = searchParams.get(UID_PARAMS.vidB) || '';
 
+  useEffect(() => {
+    if (alreadyMadeComparisons !== undefined) {
+      setComparisonsMade(alreadyMadeComparisons);
+    }
+  }, [alreadyMadeComparisons, setComparisonsMade]);
```

Comment:
If I am not mistaken the "alreadyMadeComparisons" may be empty on the first render. So the random entities could be selected before the comparisons have been fetched.

Maybe this component should render a loader in this case, and wait for the comparisons list before initializing the comparison?

Reply:
> If I am not mistaken the "alreadyMadeComparisons" may be empty on the first render. So the random entities could be selected before the comparisons have been fetched.

That's true, most of the time is empty during the first render. Good idea to add a loader.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/705/commits/08ca381787b24919047781ace366b182b13d898f.